### PR TITLE
Update doc pages after DataFusion merge

### PR DESCRIPTION
**Pull request by randerzander**
* Removes experimental from the overview page
* Initial edits for "How It Works" page

### CODE REVIEW

File: docs/source/how_does_it_work.rst [hunk 1]
```diff
@@ -18,59 +18,25 @@ No matter of via the Python API (:ref:`api`), the command line client (:ref:`cmd
...
+We also include `SQL extensions specific to Dask-SQL <https://github.com/dask-contrib/dask-sql/blob/main/dask_planner/src/parser.rs/>`_. They specify custom language features, such as the ``CREATE MODEL`` statement.
 
 3. SQL is (maybe) optimized
 ---------------------------
 
-Once the SQL string is parsed into an instance of a :class:`SqlNode` (or a subclass of it), Apache Calcite can convert it into a relational algebra and optimize it. As this is only implemented for Calcite-own classes (and not for the custom classes such as :class:`SqlCreateModel`) this conversion and optimization is not triggered for all SQL statements (have a look into :func:`Context._get_ral`).
+Once the SQL string is parsed into an instance of a :class:`SqlNode` (or a subclass of it), DataFusion can convert it into a relational algebra and optimize it. As this is only implemented for DataFusion supported syntax (and not for the custom syntax such as :class:`SqlCreateModel`) this conversion and optimization is not triggered for all SQL statements (have a look into :func:`Context._get_ral`).
```
Comment (andygrove) [inline docs/source/how_does_it_work.rst:29]: I will create a PR against this PR with some updates here. I am working on this now.

Comment (andygrove) [reply docs/source/how_does_it_work.rst:29]: Actually, probably easier just to make suggestions here.

Comment (andygrove) [reply docs/source/how_does_it_work.rst:29]: > Actually, probably easier just to make suggestions here.

That did not go well so I created a PR instead:

https://github.com/randerzander/dask-sql/pull/1

Comment (randerzander) [reply docs/source/how_does_it_work.rst]: thanks for reviewing and completing this, Andy :)

**Review by galipremsagar — APPROVED** [review]

File: docs/source/how_does_it_work.rst [hunk 2]
```diff
@@ -98,7 +72,7 @@ If you want to follow along with the steps outlined in the following, start the
 
 and enter the SQL statement above.
 
-First, the SQL is parsed by Apache Calcite and (as it is not a custom statement) transformed into a tree of relational algebra objects.
+First, the SQL is parsed by DataFusion and (as it is not a custom statement) transformed into a tree of relational algebra objects.
 
 .. code-block:: none
```
Comment (ayushdg) [inline docs/source/how_does_it_work.rst:78]:
```
Projection: #timeseries.x + #timeseries.y
  Filter: #timeseries.x > Float64(0)
    TableScan: timeseries projection=[x, y]
```

File: docs/source/how_does_it_work.rst [hunk 3]
```diff
@@ -108,7 +82,7 @@ First, the SQL is parsed by Apache Calcite and (as it is not a custom statement)
 
 The tree output above means, that the outer instance (:class:`LogicalProject`) needs as input the output of the previous instance (:class:`LogicalFilter`) etc.
 
-Therefore the conversion to python API calls is called recursively (depth-first). First, the :class:`LogicalTableScan` is converted using the :class:`rel.logical.table_scan.LogicalTableScanPlugin` plugin. It will just get the correct :class:`dask.DataFrame` from the dictionary of already registered tables of the context.
+Therefore the conversion to Python API calls is called recursively (depth-first). First, the :class:`LogicalTableScan` is converted using the :class:`rel.logical.table_scan.LogicalTableScanPlugin` plugin. It will just get the correct :class:`dask.DataFrame` from the dictionary of already registered tables of the context.
```
Comment (ayushdg) [inline docs/source/how_does_it_work.rst:85]: for the segment above: 
The tree output above means, that the outer instance (:class:`Projection`) needs as input the output of the previous instance (:class:`Filter`) etc.

**Review by ayushdg — CHANGES_REQUESTED** [review]

**Review by ayushdg — APPROVED** [review]

**Pull request merged**